src/Microsoft.ServiceFabric.Actors.Migration/MigrationWorker.cs

```diff
@@ -39,44 +39,102 @@ public MigrationWorker(KVStoRCMigrationActorStateProvider stateProvider, ActorTy
-                    var dataToMigrate = await this.GetEnumeratedDataFromKvsAsync(phase, lastUpdatedSN + 1, endSN - lastUpdatedSN);
-                    await this.stateProvider.SaveStatetoRCAsync(dataToMigrate, cancellationToken);
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await this.GetDataFromKvsAndSaveToRCAsync(phase, lastUpdatedSN + 1, endSN - lastUpdatedSN, workerIdentifier, cancellationToken);
                 }
 
+                ActorTrace.Source.WriteInfo("MigrationWorker", "Completed migration worker " + workerIdentifier.ToString() + " payload in " + phase.ToString() + " phase.");
                 await this.UpdateMetadataOnCompletion(phase, workerIdentifier, endSN);
             }
         }
 
-        private async Task<List<KeyValuePair<string, byte[]>>> GetEnumeratedDataFromKvsAsync(MigrationPhase phase, long start, long enumerationSize)
+        private async Task GetDataFromKvsAndSaveToRCAsync(MigrationPhase phase, long start, long enumerationSize, int workerIdentifier, CancellationToken cancellationToken)
         {
             var apiName = phase == MigrationPhase.Copy ? MigrationConstants.EnumeratebySNEndpoint : MigrationConstants.EnumerateKeysAndTombstonesEndpoint;
             bool includeDeletes = phase == MigrationPhase.Copy;
-            var requestserializer = new DataContractSerializer(typeof(EnumerationRequest));
-            var keyvaluepairserializer = new DataContractSerializer(typeof(List<KeyValuePair<string, byte[]>>));
+            var keyvaluepairserializer = new DataContractSerializer(typeof(List<KeyValuePair>));
+
+            var kvsApiRequest = this.CreateKvsApiRequestMessage(start, enumerationSize, includeDeletes, apiName);
+
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var response = await Client.SendAsync(kvsApiRequest, HttpCompletionOption.ResponseHeadersRead);
+                response.EnsureSuccessStatusCode();
+                using (var stream = await response.Content.ReadAsStreamAsync())
+                {
+                    using (var streamReader = new StreamReader(stream))
+                    {
+                        string responseLine = streamReader.ReadLine();
+                        cancellationToken.ThrowIfCancellationRequested();
+                        while (responseLine != null)
+                        {
+                            List<KeyValuePair> kvsData = new List<KeyValuePair>();
+                            using (Stream memoryStream = new MemoryStream())
+                            {
+                                byte[] data = Encoding.UTF8.GetBytes(responseLine);
+                                memoryStream.Write(data, 0, data.Length);
+                                memoryStream.Position = 0;
+
+                                using (var reader = XmlDictionaryReader.CreateTextReader(memoryStream, XmlDictionaryReaderQuotas.Max))
```

Comment:
You need to make changes in KVS controller as well to serialize to string rather than bytes?

Reply:
Updated the KVS API